app/models/manageiq/providers/vmware/infra_manager/inventory/persister.rb

```diff
@@ -85,6 +87,40 @@ def vms_and_templates_assign_created_on
     end
   end
 
+  def hosts_reconnect_block
+    lambda do |inventory_collection, inventory_objects_index, attributes_index|
+      relation = inventory_collection.model_class.where(:ems_id => nil)
+      return if relation.count <= 0
+
+      inventory_objects_index.each do |ref, obj|
+        record = look_up_host(relation, obj.hostname, obj.ipaddress)
+        next if record.nil?
+
+        inventory_objects_index.delete(ref)
+        hash = attributes_index.delete(ref)
+
+        record.assign_attributes(hash.except(:id, :type))
+        if !inventory_collection.check_changed? || record.changed?
+          record.save!
+          inventory_collection.store_updated_records(record)
+        end
+
+        inventory_object.id = record.id
+      end
+    end
+  end
+
+  def look_up_host(relation, hostname, ipaddr)
+    return unless ["localhost", "localhost.localdomain", "127.0.0.1"].include_none?(hostname, ipaddr)
```

Comment:
Minor but instead of the duoble negative `unless [].include_none?`, let's use the more positive `if [].include_any?`

Reply:
:+1: done